### PR TITLE
Mia modular card

### PR DESCRIPTION
**Pull request by themiachou8**
## Documentation
Newest pull request that implements modular card component popup as well as changes to the firebase for job component

## Related PRs
N/A - closed

## Screenshots

image

Screen Shot 2022-10-16 at 9 47 28 PM

Screen Shot 2022-10-16 at 9 47 39 PM

CC: @alliu879 @surajr10 @cecilia-gao3

### CODE REVIEW

**Comment by themiachou8** [comment]
Sorry I forgot to switch branches while working on the firebase!

File: src/components/JobCard/JobCard.tsx [hunk 1]
```diff
@@ -1,8 +1,53 @@
-import React from "react";
-import { Text } from "react-native";
+import { Text, View, Pressable, Modal } from "react-native";
+import styles from './CardStyles'
+import React, {useState} from "react";
 
-const JobCard = () => {
-  return <Text>hello</Text>;
-};
+const JobCard = ({title, description, employer, hours, salary, contact}:{title: string; description: string}) => {
```
Comment (alliu879) [inline src/components/JobCard/JobCard.tsx]: you should make a new props object here for the props i.e. `JobCardProps`. see this component as an example: https://github.com/calblueprint/sunday-friends-mobile/blob/main/src/components/ProfileEditModal/ProfileEditModal.tsx

File: src/firebase/firebaseApp.ts [hunk 2]
```diff
@@ -16,6 +21,11 @@ const firebaseConfig = {
 };
 
 // Initialize Firebase
-const app = initializeApp(firebaseConfig);
+const firebaseApp = firebase;
+
+if (firebaseConfig?.projectId) {
```
Comment (alliu879) [inline src/firebase/firebaseApp.ts]: what's the purpose of this code

**Review by alliu879 — CHANGES_REQUESTED** [review]
looks good! just a few nits

File: package.json [hunk 3]
```diff
@@ -46,5 +46,11 @@
     "lint:fix": "eslint --fix",
     "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
   },
+  "resolutions": {
+    "@types/react": "17.0.2",
+    "@types/react-dom": "17.0.2",
+    "graphql": "^16.5.0"
```
Comment (shannonbonet) [inline package.json:52]: What're you using `graphql` for? Doesn't seem like this is used – remove the import if so!

File: src/firebase/firebaseApp.ts (hunk 2 above)
Comment (shannonbonet) [reply src/firebase/firebaseApp.ts]: Why is it necessary to check for projectId? What is the case where `projectId` wouldn't exist that we need to account for?

**Review by shannonbonet — APPROVED** [review]
Really clean code! :) Left some questions 

I also noticed a screenshot with async storage error. Are you still getting this?

**Review by alliu879 — APPROVED** [review]

**Pull request merged**